File: backend/internal/whitenoise/domain/noise_audio.go
package domain

type NoiseAudio struct {
	ID           string `json:"id"`
	Name         string `json:"name"`
	Category     string `json:"category"`
	RequiredTier string `json:"required_tier"`
	Locked       bool   `json:"locked"`
}

var StaticList = []NoiseAudio{
	{ID: "rain", Name: "雨声", Category: "nature", RequiredTier: "glimmer"},
	{ID: "pages", Name: "翻书声", Category: "room", RequiredTier: "glimmer"},
	{ID: "wind", Name: "风声", Category: "nature", RequiredTier: "starlight"},
	{ID: "heartbeat", Name: "心跳声", Category: "body", RequiredTier: "starlight"},
}

File: app/lib/features/whitenoise/domain/noise_audio.dart
class NoiseAudio {
  const NoiseAudio({
    required this.id,
    required this.name,
    required this.icon,
    required this.audioFile,
    required this.category,
    this.requiredTier = 'glimmer',
    this.locked = false,
  });

  final String id;
  final String name;
  final String icon;
  final String audioFile;
  final String category;
  final String requiredTier;
  final bool locked;

  NoiseAudio copyWith({bool? locked}) => NoiseAudio(
        id: id,
        name: name,
        icon: icon,
        audioFile: audioFile,
        category: category,
        requiredTier: requiredTier,
        locked: locked ?? this.locked,
      );
}
